flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -536,12 +538,20 @@ private void validateClusterSpecification(ClusterSpecification clusterSpecificat
 	protected ClusterSpecification validateClusterResources(
 		ClusterSpecification clusterSpecification,
 		int yarnMinAllocationMB,
+		int yarnSchedulerMaxVcores,
 		Resource maximumResourceCapability,
 		ClusterResourceDescription freeClusterResources) throws YarnDeploymentException {
 
 		int taskManagerCount = clusterSpecification.getNumberTaskManagers();
 		int jobManagerMemoryMb = clusterSpecification.getMasterMemoryMB();
 		int taskManagerMemoryMb = clusterSpecification.getTaskManagerMemoryMB();
+		int slotsPerTaskManager = clusterSpecification.getSlotsPerTaskManager();
```

Comment:
Check first if users have specified the jm/tm vcores through `yarn.appmaster.vcores` and `yarn.containers.vcores`.

Reply:
yarn.appmaster.vcores and yarn.containers.vcores are checked in isReadyForDeployment()